llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -158,15 +166,16 @@ class SDWADstOperand : public SDWAOperand {
   DstUnused DstUn;
 
 public:
-
   SDWADstOperand(MachineOperand *TargetOp, MachineOperand *ReplacedOp,
                  SdwaSel DstSel_ = DWORD, DstUnused DstUn_ = UNUSED_PAD)
-    : SDWAOperand(TargetOp, ReplacedOp), DstSel(DstSel_), DstUn(DstUn_) {}
+      : SDWAOperand(TargetOp, ReplacedOp), DstSel(DstSel_), DstUn(DstUn_) {}
 
   MachineInstr *potentialToConvert(const SIInstrInfo *TII,
                                    const GCNSubtarget &ST,
                                    SDWAOperandsMap *PotentialMatches = nullptr) override;
   bool convertToSDWA(MachineInstr &MI, const SIInstrInfo *TII) override;
+  bool canCombineSelections(const MachineInstr &MI,
```

Comment:
Should we also check this in `SDWADstOperand::potentialToConvert` ?

Reply:
It isn't strictly necessary since there is a check at the call site (i.e. in ` SIPeepholeSDWA::run`). Seeing that this can be confusing, I have now moved all uses of `canCombineSelections` into `potentialToConvert`. Consequently, I made the function `private`. We could also inline it if you prefer, but I have a slight preference for keeping it separate - especially since there are two uses in ` SDWASrcOperand::potentialToConvert`.